README.md

```diff
@@ -1 +1,87 @@
-# aws-cloudformation
\ No newline at end of file
+# AWS Cloudformation
+
+Easily Deploy Cloudformation templates on AWS using [Serverless Components](https://github.com/serverless/components).
+
+&nbsp;
+
+- [AWS Cloudformation](#aws-cloudformation)
+    - [1. Install](#1-install)
+    - [2. Create](#2-create)
+    - [3. Configure](#3-configure)
+    - [4. Deploy](#4-deploy)
+    - [New to Components?](#new-to-components)
+
+&nbsp;
+
+
+### 1. Install
+
+```console
+$ npm install -g serverless
+```
+
+### 2. Create
+
+Just create a `serverless.yml` file
+
+```shell
+$ touch serverless.yml
+$ touch .env      # your development AWS api keys
+$ touch .env.prod # your production AWS api keys
+```
+
+the `.env` files are not required if you have the aws keys set globally and you want to use a single stage, but they should look like this.
+
+```
+AWS_ACCESS_KEY_ID=XXX
+AWS_SECRET_ACCESS_KEY=XXX
+```
+
+### 3. Configure
+
+```yml
+# serverless.yml
+
+name: my-service
+stage: dev
+
+myStack:
+  component: '@serverless/aws-cloudformation'
+  inputs:
+    template:
+      AWSTemplateFormatVersion: '2010-09-09'
+      Description: Example stack 1
+      Resources:
+        LogGroup:
+          Type: AWS::Logs::LogGroup
+          Properties:
+            LogGroupName: /log/group/one
+            RetentionInDays: 14
+      Outputs:
+        LogGroupArn:
+          Value:
+            Fn::GetAtt:
+              - LogGroup
+              - Arn
+```
+
+Inputs can contain the following properties:
+- `stackName` **[required]**. the name of the stack
+- `template` **[required]**, the template to deploy, can also be a local path to the template, e.g. `./my-template.yml`
+- `bucket`, the deployment bucket where the template is stored before deployment. If not set a new bucket will be created.
```

Comment:
does CF require the template to be stored in an S3 bucket? 🤔 ... can't we just upload the template right from the local machine?

Reply:
Yes, it can be deployed straight from the local machine, but then the size limit of the template is 51,200 bytes. If the template is uploaded to S3 and deployed from there the size limit of the template is 460,800 bytes.

Perhaps if the `bucket` is defined it uses the bucket, else the local template deployment is used. And I'll remove the automatic bucket creation.

Reply:
thanks yeah, we designed the lambda component this way too.

Reply:
Now it uses bucket only if it is defined. Otherwise, the template is included in the  CloudFormation create/update stack request.